### ISSUE TITLE
unexpected behavior with "default" script invocation and fuzzy name matching

### ISSUE THREAD

**Issue by boneskull** [kentcdodds/nps#116]
- `nps` version: 5.0.3
- `node` version: 4.3.2
- `npm` version: 2.14.12

Here's a trivial `package-scripts.js` reproducing the issue:

```javascript
exports.scripts = {
  foo: {
    'bar-baz': 'echo "foo.bar-baz"',
    bar: {
      default: 'nps foo.bar.baz',
      baz: 'echo "foo.bar.baz"'
    }
  }
};
```

The command executed:

```console
$ nps foo.bar
```

The output:

```console
nps executing: echo "foo.bar-baz"
foo.bar-baz
```

Problem description:

It appears the fuzzy script name matching is overeager when "default" scripts are invoked.

Suggested solution:

The script name matching should attempt literal keypath access *before* fuzzy matching if a "default" script is invoked.

**Comment by boneskull** [comment]
fwiw:

```shell
$ nps foo.bar.baz
```

results in:

```console
nps executing: echo "foo.bar.baz"
foo.bar.baz
```

**Comment by kentcdodds** [comment]
Yeah, this seems to be related to #24 which we kinda gave up because it's rare to happen. But, I would definitely like to see things improved!
